py3dtilers/ObjTiler/obj.py

```diff
@@ -110,22 +114,31 @@ def retrieve_objs(files, with_texture=False):
         Create Obj instance from OBJ file(s).
         :param files: paths to files
         :param with_texture: a boolean indicating if the textures should be read
-
         :return: a list of Obj.
         """
         objects = list()
 
         for obj_file in files:
             print("Reading " + str(obj_file))
-            geom = pywavefront.Wavefront(obj_file, collect_faces=True)
+            geom = pywavefront.Wavefront(obj_file, collect_faces=True, create_materials=True)
+            mesh = geom.mesh_list[0]
```

Comment:
> mesh = geom.mesh_list[0]

We need to iterate on all the meshes in the obj file, so we can delete this line (see comment below)